drivers/include/abp2.h

```diff
@@ -0,0 +1,313 @@
+ * @brief      SPI and I2C pressure and temperature sensor driver.
+ *
+ *
+ * ## Description
+ *
+ * The unit and the range depend on the part number of the sensor.
+ * It is the responsibility of the user to use homogeneous units, depending
+ * on the actual sensor being used.
+ * The pressure range is set when calling #abp2_init().
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * The functions return 0 on success or a negative value made from \c errno.h on error.
+ *
+ * ## Usage
+ *
+ * See `RIOT/tests/drivers/abp2` for an example application using this driver.
+ *
+ * Add the following modules to the Makefile:
+ * \code
+ * USEMODULE += abp2
+ * USEMODULE += abp2_spi
+ * \endcode
+ * When the I2C version is supported, `abp2_i2c` can be used for I2C sensors
+ * instead of `abp2_spi`.
+ *
+ * Initialize the driver by calling #abp2_init().
+ *
+ * When a measurement is started on the sensor, its internal ADC converts
+ * both pressure and temperature.
+ * The conversion takes about 5ms.
+ * These two values are retrieved in a single bus transfer.
+ *
+ * There are two ways to use the driver:
+ * - blocking mode : call #abp2_read()
+ * - non-blocking mode : call #abp2_read_nb()
+ *
+ * The former will block for about 5ms and return the latest data, while
+ * the latter will return almost immediately but will return data from
+ * the previous ADC conversion.
+ *
+ * ## Caveats
+ *
+ * This driver currently doesn't support the I2C flavor of these sensors.
+ *
+ * @{
+ *
+ * @file
+ * @brief      Honeywell ABP2 series pressure and temperature
+ *             sensor driver
+ *
+ * @author     David Picard <david.picard@clermont.in2p3.fr>
+ */
+
+
+#ifndef ABP2_H
+#define ABP2_H
+
+#include "saul.h"
+
+#if defined(MODULE_ABP2_SPI)
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#elif defined(MODULE_ABP2_I2C)
+#include "periph/i2c.h"
+#else
+#error "ABP2 driver >> Please add a valid module: either abp2_spi or abp2_i2c"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(MODULE_ABP2_I2C)
+/**
+ * @brief   Default I2C slave address for ABP2 devices
+ *
+ * The sensor address cannot be changed.
+ * A different address can only be obtained upon special request to Honeywell.
+ */
+#define ABP2_ADDR_DEFAULT       (0x28)
+#endif
+
+
+/**
+ * @defgroup abp2_statusbits Status bits
+ * @{
+ *  Macro definitions to manipulate the bits of the status byte.
+ *
+ * Status byte explanation :
+ *
+ *  Bit | Meaning                       | Comment |
+ *  --- | ---                           | --- |
+ *  7   | always 0                      | |
+ *  6   | 1 : device is powered         | |
+ *  5   | 1 : device is busy            | Wait longer between 2 commands. |
+ *  4   | always 0                      | |
+ *  3   | always 0                      | |
+ *  2   | 1 : integrity test failed     | Checksum error at power-up. |
+ *  1   | always 0                      | |
+ *  0   | 1 : internal math saturation  | |
```

Comment:
looks strange all but last collum are propper

Reply:
What's the problem ? Some cells in the last columns are empty, that's all.

Reply:
```suggestion
 *  Bit | Meaning                       | Comment                         |
 *  --- | ---                           | ---                             |
 *  7   | always 0                      |                                 |
 *  6   | 1 : device is powered         |                                 |
 *  5   | 1 : device is busy            | Wait longer between 2 commands. |
 *  4   | always 0                      |                                 |
 *  3   | always 0                      |                                 |
 *  2   | 1 : integrity test failed     | Checksum error at power-up.     |
 *  1   | always 0                      |                                 |
 *  0   | 1 : internal math saturation  |                                 |
```